src/index.js

```diff
@@ -1,29 +1,41 @@
-var using = function (values, func) {
-    if (values instanceof Function) {
-        values = values();
+(function (factory) {
+    'use strict';
+    if (typeof define === 'function' && define.amd) {
```

Comment:
I would stay in one definition type - here you use "typeof" and lower you use "instanceof" - use "instanceof" here as well